File: spring-ai-alibaba-deepresearch/src/main/java/com/alibaba/cloud/ai/example/deepresearch/node/annotation/Node.java
/*
 * Copyright 2025 the original author or authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.alibaba.cloud.ai.example.deepresearch.node.annotation;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * The Node annotation is used to mark a class as a node in a graph or workflow. It
 * provides metadata about the node, including its name and description. This annotation
 * can be used to facilitate the identification and documentation of nodes within a
 * system.
 *
 * @author ViliamSun
 * @since 1.0.0
 */

@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
@Documented
public @interface Node {

	String name();

	/**
	 * The description of the node, which can be used to provide additional context or
	 * information about the node's purpose. This description is typically used in user
	 * interfaces or documentation to help users understand what the node does.
	 * @return a string describing the node
	 */
	String description();

}

File: spring-ai-alibaba-deepresearch/ui-vue3/src/types/node.ts
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
export interface GraphId {
  session_id: string
  thread_id: string
}
/**
 * 普通类型节点定义
 */
export interface NormalNode {
  /** 节点名称 */
  nodeName: string
  /** 节点内容 */
  content: string | any ,
  graphId: GraphId
  displayTitle: string,
  siteInformation: SiteInformation[]
}

export interface SiteInformation {
  icon: string
  weight: string
  title: string
  url: string
  content: string
}

export interface LlmStreamNode {
  visible: boolean,
  step_title: string,
  finishReason: string
  graphId: GraphId
}
